hicommonwealth/commonwealth
# Comments Optimization

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8015 https://github.com/hicommonwealth/commonwealth/issues/10174

## Description of Changes

- Added multi-level pagination for comments and replies - Comments (both main and nested replies list) will be loaded with a "Load more" button instead of an infinite scroll. The issue with infinite scroll is it depends on list heights, and a list with 0 heights causes issues with nested/hierarchy structure loading like we have with this comments list.
- Added reply counts and comment levels to comments/+replies.
- Reorganized comment components and removed stale components
- Removed comments cache logic, to be added again in https://github.com/hicommonwealth/commonwealth/issues/10461
- Removed older `/viewComments` endpoint
- Correctly disable reply states once a certain comment reply level is reached
- a lot more minor bug and state fixes

## "How We Fixed It"
N/A

## Test Plan
1. Visit any thread that has a lot of comments ex: `stargatetoken/discussion/4353-proposal-guidelines`
2. Verify comments load in a paginated pattern
3. Verify nested replies load correctly
4. Verify nested replies on a deleted comment load correctly
5. Verify comment filters work correctly
6. Verify mutation actions (add/remove/edit comment/replies/reactions) on a comment correctly update comments tree
7. Verify you can't add more than 8 level of replies on a comment/reply tree
8. Verify you can jump highlight on a parent level comment using URL i.e share comment > visit URL > notice comment gets highlighted

## Deployment Plan
N/A

## Other Considerations
- Unrelated BUG: Thread comment count doesn't correctly update. It also counts deleted comments that have no replies. -- https://github.com/hicommonwealth/commonwealth/issues/10460
- Followup for focus highlighting comments: https://github.com/hicommonwealth/commonwealth/issues/10459
